compiler/src/dmd/dsymbolsem.d

```diff
@@ -3753,10 +3753,10 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         Linterfaces:
             bool foundVtblMatch = false;
 
-            for (ClassDeclaration bcd = cd; !foundVtblMatch && bcd; bcd = bcd.baseClass)
-            {
-                foreach (b; bcd.interfaces)
+            cd.forAllInterfaces((b)
                 {
+                    if (foundVtblMatch)
+                        return;
```

Comment:
This is a bit inefficient, can fix. There is also an existing return statement on line 3769 which currently returns from the function literal when it should return from the parent function. That might be the problem in the original pull. Will probably fix that tomorrow.